stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AndroidDOMProviderFactory.java

```diff
@@ -11,18 +11,59 @@
 
 import android.app.Application;
 
+import android.os.Handler;
+import android.os.Looper;
+import com.facebook.stetho.common.ThreadBound;
+import com.facebook.stetho.common.UncheckedCallable;
 import com.facebook.stetho.common.Util;
-import com.facebook.stetho.inspector.elements.DOMProvider;
+import com.facebook.stetho.common.android.HandlerUtil;
+import com.facebook.stetho.inspector.elements.DocumentProvider;
 
-public class AndroidDOMProviderFactory implements DOMProvider.Factory {
+public final class AndroidDOMProviderFactory
+    implements DocumentProvider.Factory, ThreadBound {
   private final Application mApplication;
+  private final Handler mHandler;
 
   public AndroidDOMProviderFactory(Application application) {
     mApplication = Util.throwIfNull(application);
+    mHandler = new Handler(Looper.getMainLooper());
   }
 
   @Override
-  public DOMProvider create() {
-    return new AndroidDOMProvider(mApplication);
+  public DocumentProvider create() {
+    return new AndroidDocumentProvider(mApplication, mHandler);
```

Comment:
I tried doing this, but there is an issue since we shouldn't override methods of `ThreadBoundProxy`, while the implementation of `postDelayed(...)` in `AndroidDocumentProvider` is different from the one in `ThreadBoundProxy`.